internal/newrelic/client.go

```diff
@@ -142,17 +146,100 @@ func nrqlQueryExpectedValueTest(nrc *nrClient, query string, expectedResults []s
 	a, _ := nrc.client.Query(nrc.accountID, query)
 
 	if len(expectedResults) != len(a.Results) {
-		return fmt.Errorf("%w: %s - expected %d got %d", ErrResultNumber, query, len(expectedResults), len(a.Results))
+		return fmt.Errorf("%w: %s\n - expected %d got %d", ErrResultNumber, query, len(expectedResults), len(a.Results))
 	}
 	for i, expectedResult := range expectedResults {
-		stringResult := fmt.Sprintf("%v", a.Results[i][expectedResult.Key])
-		if stringResult != expectedResult.Value {
-			return fmt.Errorf("%w: %s - expected for key '%s': '%s' got '%s'", ErrNotExpectedResult, query, expectedResult.Key, expectedResult.Value, stringResult)
+		actualResult := a.Results[i][expectedResult.Key]
+		comparisonErr := compareResults(actualResult, expectedResult)
+		if comparisonErr != nil {
+			return fmt.Errorf("%w: %s\n - for key '%s': %s", ErrNotExpectedResult, query, expectedResult.Key, comparisonErr.Error())
 		}
 	}
 	return nil
 }
 
+func compareResults(actualResult any, expectedResult spec.TestNRQLExpectedResult) error {
+	expectedExactResult := expectedResult.Value
+	if expectedExactResult != nil {
+		// We are checking for an exact value
+		expectedExactResult = preprocessResult(expectedExactResult)
+		actualResult = preprocessResult(actualResult)
+
+		if expectedExactResult == actualResult {
+			return nil
+		}
+		return fmt.Errorf("%w - expected: '%s', got '%s'", ErrAssertionFailure, expectedExactResult, actualResult)
+	}
+
+	// We are checking for a bounded value
+	return checkBounds(actualResult, expectedResult.LowerBoundedValue, expectedResult.UpperBoundedValue)
+}
+
+func preprocessResult(result any) any {
+	switch typedResult := result.(type) {
+	case int:
+		intResult := typedResult
+		// Convert int into floats
+		return float64(intResult)
+	case string:
+		stringResult := typedResult
+		// Convert string nil into nil
+		if strings.EqualFold(stringResult, "nil") {
+			return nil
+		}
+
+		// Convert string booleans into boolean type
+		if strings.EqualFold(stringResult, "false") {
+			return false
+		} else if strings.EqualFold(stringResult, "true") {
+			return true
+		}
+		return stringResult
+	}
+
+	return result
+}
+
+func extractFloat(result any) (float64, error) {
+	result = preprocessResult(result)
+	floatResult, ok := result.(float64)
+	if !ok {
+		return 0, fmt.Errorf("%w: float", ErrTypeAssertion)
+	}
+	return floatResult, nil
+}
+
+//nolint:gocyclo // This function is self-explanatory and simplifying it would be more convoluted
```

Comment:
^ i think this is a good suggestion, but might be a little easier to understand if written this way: 
```
// if either lowerBound is nil and result is less than upper bound, OR upperbound is nil and result is greater than lower bound, return nil, otherwise return an error
if (lowerBound == nil && *upperBound >= result) || (upperBound == nil && result <= *lowerBound) {
       return nil
}
rangeAsString := formatRange(lowerBound, upperBound) // returns "[-Inf, x]" "[x, Inf]" or "[x, y]" depending of bounds
return fmt.Errorf("%w - expected value in range %s", ErrAssertionFailure, rangeAsString)
```

Reply:
i agree w/ the suggestion above, w/ the caveat that i think it would be a little bit more straightforward if written this way, because its easier to add a comment on what the expected behavior is, (and also i think the conditional for checking lowerBound against the result was swapped, should be `<=` rather than `>=`) but its a personal preference: 
```
// if either lowerBound is nil or lowerBound <= result, AND either upperBound is nil or result <= upperBound, return nil, otherwise, error 
if (lowerBound == nil || *lowerBound <= result) && (upperBound == nil || result <= *upperBound) {
       return nil
}
rangeAsString := formatRange(lowerBound, upperBound) // returns "[-Inf, x]" "[x, Inf]" or "[x, y]" depending of bounds
return fmt.Errorf("%w - expected value in range %s", ErrAssertionFailure, rangeAsString)
```

Reply:
Looks good to me. Updated the PR.